src/Akka.Streams.Kafka/Stages/Consumers/Abstract/SingleSourceStageLogic.cs

```diff
@@ -48,27 +25,30 @@ internal class KafkaSourceStageLogic<K, V> : GraphStageLogic
         private readonly TaskCompletionSource<NotUsed> _completion;
         private readonly CancellationTokenSource _cancellationTokenSource;
 
-        public KafkaSourceStageLogic(KafkaSourceStage<K, V> stage, Attributes attributes, TaskCompletionSource<NotUsed> completion) : base(stage.Shape)
+        public SingleSourceStageLogic(SourceShape<TMessage> shape, ConsumerSettings<K, V> settings, 
```

Comment:
This source stage was transformed to more general `SingleSourceStage`, that uses `IMessageBuilder` argument to pass consumed messages to downstream and does not depend on concrete stage like (because only shape of the stage is required to implement logic, not the stage itself).